### ISSUE TITLE
[physx] build failure

### ISSUE THREAD

**Issue by omartijn** [microsoft/vcpkg#18584]
**Host Environment**
- OS: Linux (64-bit)
- Compiler: clang 11.0.1

**To Reproduce**
`./vcpkg install physx`

**Failure logs** 
```
PHYSX ROOT /home/martijn/git/vcpkg/buildtrees/physx/src/be49a99479-2db4990732/physx
PhysX Build Platform: linux
Using CXX Compiler: /usr/bin/c++
CMake Error at /home/martijn/git/vcpkg/buildtrees/physx/src/be49a99479-2db4990732/externals/cmakemodules/GetCompilerAndPlatform.cmake:126 (IF):
  if given arguments:

    "STREQUAL" "x86_64"

  Unknown arguments specified
Call Stack (most recent call first):
  /home/martijn/git/vcpkg/buildtrees/physx/src/be49a99479-2db4990732/externals/cmakemodules/NvidiaBuildOptions.cmake:99 (GetPlatformBinName)
  /home/martijn/git/vcpkg/buildtrees/physx/src/be49a99479-2db4990732/physx/source/compiler/cmake/CMakeLists.txt:70 (INCLUDE)
```

**Additional context**
The command that fails is this:

```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: /home/martijn/git/vcpkg/downloads/tools/cmake-3.20.2-linux/cmake-3.20.2-linux-x86_64/bin/cmake /home/martijn/git/vcpkg/buildtrees/physx/src/be49a99479-2db4990732.clean/physx/compiler/public -DPHYSX_ROOT_DIR=/home/martijn/git/vcpkg/buildtrees/physx/src/be49a99479-2db4990732.clean/physx -DPXSHARED_PATH=/home/martijn/git/vcpkg/buildtrees/physx/src/be49a99479-2db4990732.clean/pxshared -DPXSHARED_INSTALL_PREFIX=/home/martijn/git/vcpkg/packages/physx_x64-linux -DCMAKEMODULES_PATH=/home/martijn/git/vcpkg/buildtrees/physx/src/be49a99479-2db4990732.clean/externals/cmakemodules -DCMAKEMODULES_NAME=CMakeModules -DCMAKE_MODULES_VERSION=1.27 -DPX_BUILDSNIPPETS=OFF -DPX_BUILDPUBLICSAMPLES=OFF -DPX_FLOAT_POINT_PRECISE_MATH=OFF -DPX_COPY_EXTERNAL_DLL=OFF -DGPU_DLL_COPIED=ON -DTARGET_BUILD_PLATFORM=linux -DPX_GENERATE_STATIC_LIBRARIES=ON -DNV_USE_STATIC_WINCRT=OFF -DPX_OUTPUT_ARCH=x86 -DCMAKE_MAKE_PROGRAM=/home/martijn/git/vcpkg/downloads/tools/ninja-1.10.2-linux/ninja -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/martijn/git/vcpkg/scripts/toolchains/linux.cmake -DVCPKG_TARGET_TRIPLET=x64-linux -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/home/martijn/git/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE= -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/home/martijn/git/vcpkg -D_VCPKG_INSTALLED_DIR=/home/martijn/git/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -DPX_OUTPUT_BIN_DIR=/home/martijn/git/vcpkg/packages/physx_x64-linux/debug -DPX_OUTPUT_LIB_DIR=/home/martijn/git/vcpkg/packages/physx_x64-linux/debug -DNV_USE_DEBUG_WINCRT=ON -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/home/martijn/git/vcpkg/packages/physx_x64-linux/debug
```

I added some debugging code to `scripts/toolchains/linux.cmake` to see what those variables are set to:

`message(STATUS "*** CMAKE_SYSTEM_PROCESSOR: ${CMAKE_SYSTEM_PROCESSOR}, VCPKG_TARGET_ARCHITECTURE: ${VCPKG_TARGET_ARCHITECTURE}")`

which gives the following output in `buildtrees/detect_compiler/config-x64-linux-rel-out.log`:

`-- *** CMAKE_SYSTEM_PROCESSOR: x86_64, VCPKG_TARGET_ARCHITECTURE: x64`

Yet it fails when doing the following `IF` in `GetCompilerAndPlatform.cmake`:

`IF(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64")`

I'm stumped.

**Comment by Hoikas** [comment]
Try patching the failing line to `IF(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")`

**Comment by omartijn** [comment]
That'll get the build to not fail, but it doesn't solve the underlying issue, namely that the variable is unset. With the `x64-linux` triplet it _should_ be set to `x86_64` and it should go into that branch.

**Comment by JackBoosY** [comment]
Should add a community triplet named `x64-linux-clang` then handle the toolchain.

**Comment by omartijn** [comment]
Can you tell me a little bit more about this, @JackBoosY ? I've tried switching compilers to gcc, but that didn't seem to solve the issue (which makes sense since it's a problem with things not propagating properly through the vcpkg toolchain).

**Comment by JackBoosY** [comment]
@omartijn  Can you please add `--debug` to the install command then provide the output?

**Comment by omartijn** [comment]
I did that, and then the issue no longer appeared. I had pulled the latest master, re-run the bootstrap.sh script and cleared all cache. Now it doesn't happen even without debug.

No idea what has changed, exactly, and it wasn't some weird system configuration I had, since it happened in a CI pipeline as well: https://dev.azure.com/vcpkg/public/_build/results?buildId=54876&view=artifacts&pathAsName=false&type=publishedArtifacts
